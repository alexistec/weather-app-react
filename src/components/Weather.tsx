import React from 'react';
import Avatar from '@material-ui/core/Avatar';

type WeatherProps = {
    city:string;
    icon:string;
    humedity:number;
    temp:number
}

const Weather:React.FC<WeatherProps>= ({city,icon,humedity,temp}) => {

    return(
        <div style={{justifyContent:'center',alignItems:'center'}}>
            <h1 style={{textAlign:'center'}}>Weather in {city} </h1>
            <div style={{display: 'flex',alignItems: 'center',justifyContent: 'center'}}>
                <Avatar style={{height:100,width:100}}alt="Remy Sharp" src={`https://openweathermap.org/img/wn/${icon}.png`}  />
                <h1>{temp}Â°</h1>
            </div>
            <h3 style={{textAlign:'center'}}>Humedity {humedity}</h3>
        </div>
    )
}


export default Weather;