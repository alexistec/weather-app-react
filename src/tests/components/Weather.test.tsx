import React from "react";
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Weather from "../../components/Weather";


describe('Test in <WeatherApp/>', () => {
    const props = {
        city: 'formosa',
        icon: 'icon.jpg',
        temp: 12,
        humedity:58 
    }

    test('should renders correctly', () => {
        const wrapper = renderer.create(<Weather {...props}/>)
                        .toJSON();
        expect( wrapper ).toMatchSnapshot();
    })
    
    test('should show value props', () => {
        const { city,temp,humedity, icon } = props;
        const wrapper = render(<Weather {...props}/>);

        const displayedImage = document.querySelector("img") as HTMLImageElement;
        
        expect( wrapper.container ).toHaveTextContent( `Weather in ${city}` );       
        expect( wrapper.container ).toHaveTextContent( `${temp}Â°` );    
        expect( wrapper.container ).toHaveTextContent( `Humedity ${humedity}` );      
        
        expect(displayedImage.alt).toContain(icon)
    })
    

})
