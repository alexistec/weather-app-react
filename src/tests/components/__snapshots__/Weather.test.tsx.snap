// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <WeatherApp/> should renders correctly 1`] = `
<div
  style={
    Object {
      "alignItems": "center",
      "justifyContent": "center",
    }
  }
>
  <h1
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    Weather in 
    formosa
     
  </h1>
  <div
    style={
      Object {
        "alignItems": "center",
        "display": "flex",
        "justifyContent": "center",
      }
    }
  >
    <div
      className="MuiAvatar-root MuiAvatar-circular"
      style={
        Object {
          "height": 100,
          "width": 100,
        }
      }
    >
      <img
        alt="icon.jpg"
        className="MuiAvatar-img"
        src="https://openweathermap.org/img/wn/icon.jpg.png"
      />
    </div>
    <h1>
      12
      Â°
    </h1>
  </div>
  <h3
    style={
      Object {
        "textAlign": "center",
      }
    }
  >
    Humedity 
    58
  </h3>
</div>
`;
